javascript/apps/taiga/src/app/modules/project/feature-kanban/components/delete-status/delete-status.component.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * Copyright (c) 2023-present Kaleidos INC
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  OnInit,
+} from '@angular/core';
+import { TranslocoModule } from '@ngneat/transloco';
+import { TuiAutoFocusModule } from '@taiga-ui/cdk';
+import { TuiButtonModule, TuiLinkModule } from '@taiga-ui/core';
+import { Status } from '@taiga/data';
+import { ContextNotificationModule } from '@taiga/ui/context-notification/context-notification.module';
+import { ModalModule } from '@taiga/ui/modal';
+import { InputsModule } from '@taiga/ui/inputs/inputs.module';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { CommonModule } from '@angular/common';
+
+@Component({
+  selector: 'tg-delete-status',
+  templateUrl: './delete-status.component.html',
+  styleUrls: ['./delete-status.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: true,
+  imports: [
+    TranslocoModule,
+    TuiButtonModule,
+    TuiLinkModule,
+    ModalModule,
+    TuiAutoFocusModule,
+    ContextNotificationModule,
+    InputsModule,
+    CommonModule,
+  ],
+})
+export class DeleteStatusComponent implements OnInit {
+  @Input({ required: true })
+  public currentStatus!: Status;
+
+  @Input({ required: true })
+  public statuses!: Status[];
+
+  @Input({ required: true })
+  public isLastStatus!: boolean;
+
+  @Output()
+  public closeModal = new EventEmitter<void>();
+
+  @Output()
+  public submitDelete = new EventEmitter<Status['slug'] | undefined>();
+
+  @Input()
+  public set show(show: boolean) {
+    this.showDeleteStatusModal = show;
+  }
+
+  public showDeleteStatusModal = false;
+  public form!: FormGroup;
+
+  public get filteredStatus(): Status[] {
+    return this.statuses.filter((it) => it.name !== this.currentStatus.name);
+  }
+
+  constructor(public fb: FormBuilder) {}
+
+  public ngOnInit() {
+    if (!this.isLastStatus) {
+      this.form = this.fb.group({
+        stories: ['move'],
+        status: [this.filteredStatus[0].name],
+      });
+    }
+  }
+
+  public trackByIndex(index: number) {
+    return index;
+  }
+
+  public submit() {
+    this.close();
+    const moveToStatus: Status['slug'] | undefined =
+      !this.isLastStatus && this.form.get('stories')!.value === 'move'
+        ? (this.form.get('status')!.value as Status['slug'])
```

Comment:
Here you are using slugs as values but in the template name is used, and it is not always the same